### PR TITLE
PICARD-867: Improve id3 tag deletion

### PR DESCRIPTION
**Pull request by samj1912**

### CODE REVIEW

**Comment by samj1912** [comment]
So, while I was testing MP# tag deletion manually I noticed a lot of tags weren't being deleted. 

I then noticed I had settled for a very simple criteria for id3 tag deletion.

Improved it a lot and tested it in UI for all tracks. Works flawlessly!

Though there might still be bugs with id3 compatibility which I have not tested.

File: picard/formats/id3.py [hunk 1]
```diff
@@ -409,14 +409,64 @@ def _save(self, filename, metadata):
...
-            elif real_name in tags:
-                del tags[real_name]
+        for name in metadata.deleted_tags:
+            real_name = self._get_tag_name(name)
+            try:
+                if name.startswith('performer:'):
+                    role = name.split(':', 1)[1]
+                    for people in tags['TMCL'].people:
+                        if people[0] == role:
+                            del people
```
Comment (mwiencek) [inline picard/formats/id3.py]: Does that do what you expect?

File: picard/formats/id3.py [hunk 2]
```diff
@@ -409,14 +409,64 @@ def _save(self, filename, metadata):
...
+                    else:
+                        desc = ''
+                    for key, frame in tags.items:
+                        if frame.FrameID == desc:
+                            del tags[key]
+                elif name in self._rtipl_roles:
+                    role = self._rtipl_roles[name]
+                    for people in tags['TIPL'].people:
+                        if people[0] == role:
+                            del people
```
Comment (mwiencek) [inline picard/formats/id3.py]: Same here, this looks fishy.

File: picard/formats/id3.py [hunk 3]
```diff
@@ -409,14 +409,64 @@ def _save(self, filename, metadata):
...
+                    if desc.lower()[:4] != 'itun':
+                        for key, frame in tags.items():
+                            if frame.FrameID == 'COMM' and frame.desc == desc:
+                                del tags[key]
+                elif name.startswith('lyrics:') or name == 'lyrics':
+                    if ':' in name:
+                        desc = name.split(':', 1)[1]
+                    else:
+                        desc = ''
+                    for key, frame in tags.items:
```
Comment (mwiencek) [inline picard/formats/id3.py]: `tags.items()`?

Comment (samj1912) [inline picard/formats/id3.py]: Yes. Will update.

**Comment by samj1912** [comment]
Fixed. Deletes as expected.

**Comment by samj1912** [comment]
Any update on the PR?

**Comment by zas** [comment]
It looks ok to me, i would like tests to be updated though so they cover this.

**Comment by samj1912** [comment]
Done :)

**Pull request merged**